jsonschema-plugin/src/test/groovy/io/micronaut/gradle/jsonschema/JsonSchemaGeneratorSpec.groovy

```diff
@@ -0,0 +1,213 @@
+            }
+
+        """
+
+
+        when:
+        def result = build('test')
+
+        then:
+        result.task(":generatingSourcesFromFhir").outcome == TaskOutcome.SUCCESS
+        result.task(":compileJava").outcome == TaskOutcome.SUCCESS
+
+        and:
+        file("build/generated/jsonschema/generatingSourcesFromFhir/src/main/java/io/micronaut/jsonschema/generated").exists()
+        file("build/generated/jsonschema/generatingSourcesFromFhir/src/main/java/io/micronaut/jsonschema/generated").list().size() == 838
+        file("build/generated/jsonschema/generatingSourcesFromFhir/src/main/java/io/micronaut/jsonschema/generated/Fhir.java").exists()
+    }
+
+    def "can generate source code from URL with output settings"() {
+        given:
+        allowSnapshots = true
+        settingsFile << "rootProject.name = 'jsonschema-url'"
+        buildFile << """
+            plugins {
+                id "io.micronaut.minimal.application"
+                id "io.micronaut.jsonschema"
+            }
+            
+            micronaut {
+                version "$micronautVersion"
+                jsonschema {
+                    url("https://www.hl7.org/fhir/fhir.schema.json") {
+                        lang = "java"
+                        outputPackageName = "com.example.fhir"
+                        outputDirectory = project.getLayout().getBuildDirectory().dir("generated")
+                    }
+                }
+            }
+            
+            $repositoriesBlock
+
+            dependencies {
+
+                annotationProcessor "io.micronaut.serde:micronaut-serde-processor"
+                annotationProcessor "io.micronaut.validation:micronaut-validation-processor"
+
+                implementation "io.micronaut.serde:micronaut-serde-jackson"
+                implementation "io.micronaut.validation:micronaut-validation"
+            }
+
+        """
+
+
+        when:
+        def result = build('test')
+
+        then:
+        result.task(":generatingSourcesFromFhir").outcome == TaskOutcome.SUCCESS
+        result.task(":compileJava").outcome == TaskOutcome.SUCCESS
+
+        and:
+        file("build/generated/generatingSourcesFromFhir/src/main/java/com/example/fhir/").exists()
+        file("build/generated/generatingSourcesFromFhir/src/main/java/com/example/fhir/").list().size() == 838
+        file("build/generated/generatingSourcesFromFhir/src/main/java/com/example/fhir/Fhir.java").exists()
+    }
+
+    def "can generate sources from a local file "() {
+        given:
+        allowSnapshots = true
+        settingsFile << "rootProject.name = 'jsonschema-url'"
+        buildFile << """
+            plugins {
+                id "io.micronaut.minimal.application"
+                id "io.micronaut.jsonschema"
+            }
+            
+            micronaut {
+                version "$micronautVersion"
+                jsonschema {
+                    file(file("animal.schema.json")) {
```

Comment:
nit: The double `file` was a bit confusing in the beginning. Maybe we should rename the method to `fromFile`?